docs/content/latest/deploy/kubernetes/clients.md

```diff
@@ -109,3 +109,137 @@ Status:
         router: no errors
         loadbalancer emulator: no errors
 ```
+
+## Connecting TLS Secured YugabyteDB cluster deployed by Helm Charts
+
+### Within Kubernetes cluster
+
+Copy the following `yb-client.yaml` and use this `kubectl create -f yb-client.yaml` command to create a pod with auto-mounted client certificates.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: yb-client
+  namespace: yb-demo
+spec:
+  containers:
+  - name: yb-client
+    image: yugabytedb/yugabyte-client:latest
+    env:
+    - name: SSL_CERTFILE
+      value: "/root/.yugabytedb/root.crt"
+    volumeMounts:
+    - name: yugabyte-tls-client-cert
+      mountPath: "/root/.yugabytedb/"
+  volumes:
+  - name: yugabyte-tls-client-cert
+    secret:
+      secretName: yugabyte-tls-client-cert
+      defaultMode: 256
+```
+
+Here is an example of a client that uses the `YSQL shell` ([`ysqlsh`](../../../admin/ysqlsh)) to connect.
+
+Use the following command to verify the connection.
+
+```sh
+$ kubectl exec -n yb-demo -it yb-client -- ysqlsh -h yb-tservers.yb-demo.svc.cluster.local "sslmode=require" -c \\conninfo
+You are connected to database "yugabyte" as user "yugabyte" on host "yb-tservers.yb-demo.svc.cluster.local" at port "5433".
+SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
+```
+
+Use the following command to use YugabyteDB cluster.
+
+```sh
+$ kubectl exec -n yb-demo -it yb-client -- ysqlsh -h yb-tservers.yb-demo.svc.cluster.local "sslmode=require"
+ysqlsh (11.2-YB-2.1.5.0-b0)
+SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
+Type "help" for help.
+
+yugabyte=#
+```
+
+Here is an example of a client that uses the `YCQL shell` ([`ycqlsh`](../../../admin/cqlsh)) to connect.
+
+Use the following command to verify the connection.
+
+```sh
+$ kubectl exec -n yb-demo -it yb-client -- ycqlsh yb-tservers.yb-demo.svc.cluster.local 9042 --ssl --execute 'SHOW HOST'
+Connected to local cluster at yb-tservers.yb-demo.svc.cluster.local:9042.
+```
+
+Use the following command to use YugabyteDB cluster.
+
+```sh
+$ kubectl exec -n yb-demo -it yb-client -- ycqlsh yb-tservers.yb-demo.svc.cluster.local 9042 --ssl
+Connected to local cluster at yb-tservers.yb-demo.svc.cluster.local:9042.
+[cqlsh 5.0.1 | Cassandra 3.9-SNAPSHOT | CQL spec 3.4.2 | Native protocol v4]
+Use HELP for help.
+cqlsh>
+```
+
+(Optional) Use the below-mentioned command to remove the `yb-client` pod.
+
+```sh
+$ kubectl delete po yb-client -n yb-demo
+pod "yb-client" deleted
+```
+
+### Connecting Remote Cluster
```

Comment:
`Connecting externally` might be a better name for this section.

Reply:
Yeah, this one is comparatively better. I'll update it.